state/migration_import.go

```diff
@@ -2855,20 +2851,74 @@ func (i *importer) remoteSecrets() error {
 
 func (i *importer) virtualHostKeys() error {
 	i.logger.Debugf("importing virtual host key")
-	migration := &ImportStateMigration{
-		src: i.model,
-		dst: i.st.db(),
-	}
-	migration.Add(func() error {
-		m := ImportVirtualHostKeys{}
-		return m.Execute(stateModelNamspaceShim{
-			Model: migration.src,
-			st:    i.st,
-		}, migration.dst)
-	})
-	if err := migration.Run(); err != nil {
-		return errors.Trace(err)
+
+	vhKeys := i.model.VirtualHostKeys()
+	// Generate virtual host keys when migrating from an old controller.
+	if len(vhKeys) == 0 {
+		if err := i.generateMissingVirtualHostKeys(); err != nil {
+			return errors.Trace(err)
+		}
```

Comment:
Exit early here - reduces indentation etc.

Reply:
I wanted to but didn't for probably a silly reason. There is a debug log `i.logger.Debugf("importing virtual host key succeeded")` at the bottom that I didn't want to miss/duplicate. In any case, I've just copied the log line and returned early.